src/main/java/net/neoforged/neoforge/event/village/VillagerTradesEvent.java

```diff
@@ -28,10 +29,12 @@
 public class VillagerTradesEvent extends Event {
     protected Int2ObjectMap<List<ItemListing>> trades;
     protected VillagerProfession type;
+    private final RegistryAccess registryAccess;
 
-    public VillagerTradesEvent(Int2ObjectMap<List<ItemListing>> trades, VillagerProfession type) {
+    public VillagerTradesEvent(Int2ObjectMap<List<ItemListing>> trades, VillagerProfession type, RegistryAccess registryAccess) {
```

Comment:
The previous constructor needs to be retained and marked as deprecated with a comment since it's not marked as internal and therefore has to be considered API in terms of breaking changes. It should delegate to the new constructor with `RegistryAccess.EMPTY`.